redis/src/types.rs

```diff
@@ -1147,13 +1160,24 @@ impl FromRedisValue for bool {
     }
 }
 
+impl FromRedisValue for CString {
+    fn from_redis_value(v: &Value) -> RedisResult<CString> {
+        match *v {
+            Value::Data(ref bytes) => Ok(CString::new(bytes.clone())?),
+            Value::Okay => Ok(CString::new("OK")?),
+            Value::Status(ref val) => Ok(CString::new(val.as_bytes())?),
+            _ => invalid_type_error!(v, "Response type not string compatible."),
+        }
+    }
+}
+
 impl FromRedisValue for String {
     fn from_redis_value(v: &Value) -> RedisResult<String> {
         match *v {
             Value::Data(ref bytes) => Ok(from_utf8(bytes)?.to_string()),
             Value::Okay => Ok("OK".to_string()),
             Value::Status(ref val) => Ok(val.to_string()),
-            _ => invalid_type_error!(v, "Response type not string compatible."),
+            _ => invalid_type_error!(v, "Response type not CString compatible."),
```

Comment:
This is the wrong way around, you have `CString` in the error message for the `String` impl and `string` in the `CString` impl.

Reply:
🤦
Fixed